src/schema.py

```diff
@@ -15,11 +16,14 @@
 
 model_schema = {
     "family": merge(tstring, allowed(SEQ_MODELS)),
-    "n_positions": merge(tinteger, required),  # maximum context length
-    "n_dims": merge(tinteger, required),  # latent dimension
-    "n_embd": merge(tinteger, required),
-    "n_layer": merge(tinteger, required),
-    "n_head": merge(tinteger, required),
+    "n_positions": merge(tinteger, nullable, default(None)),  # maximum context length
+    "n_dims": merge(tinteger, nullable, default(None)),  # latent dimension
+    "n_embd": merge(tinteger, nullable, default(None)),
+    "n_layer": merge(tinteger, nullable, default(None)),
+    "n_head": merge(tinteger, nullable, default(None)),
+    "context_len" : merge(tinteger, nullable, default(None)),
+    "hidden_dimensions" : merge(tlist, nullable, default(None)),
+    "activation" : merge(tstring, nullable, default(None))
```

Comment:
This paradigm of stripping away the Cerebrus checking doesn't feel great, but idk a better method yet. Thoughts?

Reply:
I'm not sure what you mean; I'm not familiar with Cerberus checking. What does Cerberus test?

Reply:
Cerberus is a schema validator - - basically typechecking and nullchecking for JSON/YAML/python dicts. What I changed this from is a bunch of non-nullable fields that don't work with MLP to a bunch of nullable fields (i.e. removing the checking) that also support the MLP version. 

I added a custom null check called `NULLCHK` in `src/consts.py` that I use where necessary (i.e. in the `TransformerModel` base class and in `MLPSequence` in `src/mlp_seq.py`.

Reply:
What about this process did you not like? Why did you want a better method and would that method do?

Reply:
Ok wait never mind I understand this now and the use of the NULL_CHK function